### ISSUE TITLE
R1 S/N:003 – Errors with R1 hands (ergomorphing)

### ISSUE THREAD

**Issue by fbrand-new** [robotology/icub-tech-support#1989]
### Robot Name 🤖

R1 S/N:003

### Request/Failure description

I just turned on the robot with the new hands calibrated by @MSECode and I have these errors:
![Image](https://github.com/user-attachments/assets/ca5ccb5b-b7af-47d5-b74b-61d1bf972ff3)
I do not know if it is safe to proceed further, or if I should turn off the roboto

@AntonioConsilvio @Fabrizio69 @MSECode 

### Detailed context

See above

**Comment by MSECode** [comment]
Hi,
can u send me the configuration file of the calibrator and  the one of the motion control, that I'll double check the calibration parameters and the PID values. 
Thanks

**Comment by fbrand-new** [comment]
Update with the log files of the yri from inside and outside the docker on the base
[log.txt](https://github.com/user-attachments/files/18061235/log.txt)
[log_no_docker.txt](https://github.com/user-attachments/files/18061236/log_no_docker.txt)

**Comment by mfussi66** [comment]
The OP issue regarding i2t limit seems a side-effect of the calibration procedure being unable to read the FAP data. The sensors read just fine and we can see it in the port `/cer/left_hand/POS<2 or 4>:o` if the service is enabled in the yarprobotinterface config file.

**Comment by maggia80** [comment]
- embobjPOS was not enabled in CCMake in the r1-base  
- POS services were not enabled in the yarprobotinterface.ini file 

we still have issues in the right hand:

 ![Image](https://github.com/user-attachments/assets/48156470-5858-4b64-af42-08832d5981d2)

**Comment by MSECode** [comment]
> In order to run the calibration procedure correctly, as I talked with [@MSECode](https://github.com/MSECode) , the POS service needs to **be active** before the calibrator. If the opposite happens, the operation fails.
> 
> In ergoCubSN002 the POS service starts right at the beginning of the yarprobotinterface file:
> 
> https://github.com/robotology/robots-configuration/blob/master/ergoCubSN002/ergocub.xml#L12-L22 .
> 
> Instead, in R1SN003 it starts later:
> 
> https://github.com/robotology/robots-configuration/blob/master/R1SN003/R1.xml#L36-L46
> 
> and therefore the time between the POS and the calibrator is not sufficient.
> 
> I'm not entirely sure how the yarprobotinterface works, but I'm afraid there is some kind of race condition in this scenario.

Exactly, or at least that is what I've always seen on `ergoCub` and `iCub` robots, with the POS service running first. Thus, that can be said to be the standard use case. I'm not sure, anyways, if that is mandatory for having the whole system running correctly but what @mfussi66  said is totally correct.

**Comment by randaz81** [comment]
@mfussi66 @AntonioConsilvio can you provide and update on this activity?

**Comment by mfussi66** [comment]
Hi @randaz81 , 
The issue was solved by:

- Enforcing the priority at startup of of the POS service over the calibrators
- Tuning the PID controllers

I'll do a PR in robots-configuration ASAP.

I'm not sure if some mechanical maintainance is required, @AntonioConsilvio might know more about this.

**Comment by mfussi66** [comment]
PR here: https://github.com/robotology/robots-configuration/pull/705

The right ring+pinky is not moving correctly, @AntonioConsilvio is on it.

**Comment by edersonferrante** [comment]
- [x] We found the broken pinky phalanx and replaced it ( Cod. 17391)

 ![Image](https://github.com/user-attachments/assets/98bb38f8-3ba2-4917-b86e-f5887e8cf758)

https://github.com/user-attachments/assets/5d8a22e4-800d-477a-9db6-b539ce1dbce7
@Fabrizio69

**Comment by AntonioConsilvio** [comment]
Here the PR with the new calibration of right pinky:

- https://github.com/robotology/robots-configuration/pull/707

I tested the robot and it works properly. 

Now I am going to close this issue ✅. If other problems appear, feel free to open new issues!

**Comment by lornat75** [comment]
thanks!
